### DISCUSSION TITLE
Kettler Racer S support

### DISCUSSION THREAD

**Discussion by dvmarinoff** [dvmarinoff/Auuki#60] [Trainer Support]
The Kettler Racer S uses a custom bluetooth service for trainer control. There is a known implementation at [github.com/kaegi/kdri](https://github.com/kaegi/kdri) written in Rust with BlueZ.

Suggested list of tasks:
- [ ] creating a spec document, similar to FTMS and ANT+ FE-C, with bluetooth service id, supported operations, values and their formats. This could be added to the repo [Wiki](https://github.com/dvmarinoff/Flux/wiki/Kettler-Trainer-Control).

**Comment by bastidest** [comment]
I've been toying around with the current code and I have familiarized myself with how WebBluetooth works. The Kettler Racer S uses a plain-old binary bidirectional communication channel (RFCOMM) to transmit data according to the proprietary protocol. After reading through the WebBluetooth standard proposals, I noticed that RFCOMM channels are [out of scope](https://github.com/WebBluetoothCG/web-bluetooth/blob/main/charter.md#out-of-scope), as only Bluetooth versions >4 are targeted by WebBluetooth.

Therefore the protocol cannot be implemented using WebBluetooth. The only workaround I can think of would be communicating with a generic Bluetooth adapter using WebSerial. This feels like it goes beyond the scope of this project as very few trainers require RFCOMM channels.

**Reply by dvmarinoff** [reply]
It’s getting complicated, but if you feel like tinkering with web-serial, given that there is a good chance it might not work or be very limited in terms of platform or feature support, I don’t mind including the result in the project.

**Reply by dvmarinoff** [reply]
They've just added this to the web-serial [spec](https://github.com/WICG/serial/pull/150) I don't know if it can help you.
